src/ansibleCompletionItemProvider.ts

```diff
@@ -36,7 +36,7 @@ export class AnsibleCompletionItemProvider implements vscode.CompletionItemProvi
     }
 
     private enableAutoCompletion(document: vscode.TextDocument): boolean {
-        if (!utilities.getCodeConfiguration('ansible', 'autocompletion')) {
+        if (document.languageId == 'yaml' && !utilities.getCodeConfiguration('ansible', 'autocompletion')) {
```

Comment:
But you added the `language: 'ansible'`, here we only check yaml?

Reply:
yes. if user set language id ='ansible', then they want ansible  experience, so auto completion is enabled by default, and no configurable. if user set language id='yaml', then auto completion is optional, they have configuration to enable/disable it per file.